src/main/java/se/sundsvall/ai/flow/service/Executor.java

```diff
@@ -13,27 +13,29 @@
 import org.springframework.stereotype.Service;
 import org.zalando.problem.Problem;
 import org.zalando.problem.Status;
-import se.sundsvall.ai.flow.integration.intric.IntricIntegration;
+import se.sundsvall.ai.flow.integration.intric.IntricService;
 import se.sundsvall.ai.flow.model.flowdefinition.FlowInputRef;
 import se.sundsvall.ai.flow.model.flowdefinition.RedirectedOutput;
 import se.sundsvall.ai.flow.model.session.Input;
 import se.sundsvall.ai.flow.model.session.Session;
 import se.sundsvall.ai.flow.model.session.StepExecution;
 import se.sundsvall.ai.flow.model.support.StringMultipartFile;
+import se.sundsvall.dept44.requestid.RequestId;
 
 @Service
 public class Executor {
 
 	private static final Logger LOG = LoggerFactory.getLogger(Executor.class);
 
-	private final IntricIntegration intricIntegration;
+	private final IntricService intricService;
 
-	public Executor(final IntricIntegration intricIntegration) {
-		this.intricIntegration = intricIntegration;
+	public Executor(final IntricService intricService) {
+		this.intricService = intricService;
 	}
 
 	@Async
-	public void executeSession(final Session session) {
+	public void executeSession(final Session session, final String requestId) {
+		RequestId.init(requestId);
```

Comment:
Put all in a try {} finally {} and add RequestId.reset() in the finally section?